backend/src/main/java/com/sscatalog/specialistsservicescatalog/services/PaymentsService.java

```diff
@@ -0,0 +1,56 @@
+package com.sscatalog.specialistsservicescatalog.services;
+
+import com.sscatalog.specialistsservicescatalog.config.properties.PaymentsProperties;
+import com.sscatalog.specialistsservicescatalog.dtos.MakeStripePaymentRequest;
+import com.sscatalog.specialistsservicescatalog.entities.OfferedService;
+import com.sscatalog.specialistsservicescatalog.entities.ServiceRequest;
+import com.sscatalog.specialistsservicescatalog.entities.ServiceRequestStatus;
+import com.sscatalog.specialistsservicescatalog.exceptions.ApiException;
+import com.sscatalog.specialistsservicescatalog.repositories.ServiceRequestRepository;
+import com.stripe.Stripe;
+import com.stripe.exception.StripeException;
+import com.stripe.model.Charge;
+import com.stripe.param.ChargeCreateParams;
+import org.springframework.stereotype.Service;
+
+@Service
+public class PaymentsService {
+
+    private final ServiceRequestRepository serviceRequestRepository;
+
+    public PaymentsService(ServiceRequestRepository serviceRequestRepository, PaymentsProperties paymentsProperties) {
+        Stripe.apiKey = paymentsProperties.getStripeSecretKey();
+        this.serviceRequestRepository = serviceRequestRepository;
+    }
+
+    public void makeStripePayment(MakeStripePaymentRequest request) {
+        ServiceRequest serviceRequest = serviceRequestRepository.findById(request.getServiceRequestId())
+                                                                .orElseThrow(() -> new ApiException(
+                                                                        "Service request not found"));
+        if (serviceRequest.isPaid() || serviceRequest.getStatus() != ServiceRequestStatus.FINISHED) {
+            throw new ApiException("Service request is not payable");
+        }
+        OfferedService requestedService = serviceRequest.getRequestedService();
+        ChargeCreateParams chargeParams = ChargeCreateParams.builder()
+                                                            .setAmount(toStripePrice(requestedService.getPrice()))
+                                                            .setCurrency("USD")
+                                                            .setSource(request.getToken()
+                                                                              .getId())
+                                                            .build();
+        Charge charge;
```

Comment:
We can extract the payment in a separate helper method (lines 34-48) (it could be named 'pay(requestedServicePrice, requestTokenId)')